src/api/goods.ts

```
@@ -7,10 +7,17 @@ export function getAll(): Promise<Good[]> {
   return fetch(API_URL).then(response => response.json());
 }
 
-export const get5First = () => {
-  return getAll().then(goods => goods); // sort and get the first 5
-};
+export function get5First(): Promise<Good[]> {
+  return getAll().then(
+    goods =>
+      [...goods]
+        .sort((a, b) => a.name.localeCompare(b.name))
+        .slice(0, 5),
+  );
+}
 
-export const getRedGoods = () => {
-  return getAll().then(goods => goods); // get only red
-};
+export function getRedGoods(): Promise<Good[]> {
+  return getAll().then(
+    goods => goods.filter(good => good.color === 'red'),
+  );
+}
```

`getRedGoods` relies on `getAll` but does not add error handling or add context to propagated errors. The checklist requires that requests handle possible errors: "3. [API] - When you make a request to the server, don't forget to handle possible errors". Ensure errors from `getAll` propagate to the UI (or add catch/rethrow with clearer messages) and that the UI handles them. Do not swallow errors silently.